clmm/theory/ccl.py

```diff
@@ -152,5 +152,52 @@ def _eval_excess_surface_density(self, r_proj, z_cl):
         else:
             return self.eval_mean_surface_density(r_proj, z_cl) - self.eval_surface_density(r_proj, z_cl)
 
-
+    def _prepare_ccl(self, r_proj, z_cl, z_src):
+        """Function that Prepares the lists for the functions below"""
+        r_proj_l = None
+        z_src_l = None
+        if type(r_proj) == float:
+            r_proj_l = [r_proj]
+        else:
+            r_proj_l = r_proj
+        if type(z_src) == float:
+            z_src_l = [z_src for a in range(0, len(r_proj))]
+        else:
+            z_src_l = z_src
+        a_src = [self.cosmo.get_a_from_z(reds_src) for reds_src in z_src_l]
+        a_cl = self.cosmo.get_a_from_z(z_cl)
+        ra_proj = [radius/a_cl for radius in r_proj_l]
+        return [ra_proj, a_cl, a_src]
+    
+    def _eval_convergence(self, r_proj, z_cl, z_src):
+        """eval convergence"""
+        ra_proj, a_cl, a_src =self._prepare_ccl(r_proj, z_cl, z_src)        
+        func = lambda ra_proj, a_cl, a_src: self.hdpm.convergence(self.cosmo.be_cosmo, ra_proj, self.mdelta, a_cl, a_src, self.mdef)
+        
+        return np.vectorize(func)(ra_proj, a_cl, a_src)
```

Comment:
Lastly, I use np.vectorize so these functions work the same way as the NumCosmo backend.